### PR TITLE
fixes #18533 - container name validation

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

File: test/models/repository_test.rb [hunk 1]
```diff
@@ -78,28 +78,21 @@ def test_docker_repository_docker_upstream_name_format
...
-      refute @repo.valid?
-      @repo.docker_upstream_name = '/valid'
-      refute @repo.valid?
-      @repo.docker_upstream_name = 'thisisnotvalidbecauseitistoolong/valid'
-      refute @repo.valid?
-      @repo.docker_upstream_name = 'valid/thisisnotvalidbecauseitistoolong'
-      refute @repo.valid?
+      %w( valid abc/valid
+          thisisareallylongbutstillvalidname soisthis/thisisareallylongbutstillvalidname
+          single/slash ).each do |name|
```
Comment (ehelms) [inline test/models/repository_test.rb]: This test would be a bit easier to read if this list and the one below were named arrays before being used. e.g

```
valid = %w()
invalid = %w()
```

File: test/models/repository_test.rb [hunk 2]
```diff
@@ -78,28 +78,21 @@ def test_docker_repository_docker_upstream_name_format
...
+      end
+      %w( things\ with\ spaces UPPERCASE Uppercase uppercasE $ymbols
+          Upper/case UPPER/case upper/Case
+          $tuff.th@t.m!ght.h@ve.w%!rd.r#g#x.m*anings()
+          /startingslash trailingslash/
+          abcd/.-_
+          multiple/slash/es abc/def/invalid ).each do |name|
+        @repo.docker_upstream_name = name
+        refute(@repo.valid?, "container image name is not valid '#{name}'")
+      end
```
Comment (johnpmitsch) [inline test/models/repository_test.rb]: should we test one that is too long as well?

**Comment by thomasmckay** [comment]
@ehelms @johnpmitsch - updated tests

**Comment by johnpmitsch** [comment]
@thomasmckay should we test a name that is >255 characters fails?

**Comment by thomasmckay** [comment]
@johnpmitsch https://github.com/Katello/katello/pull/6660/files#diff-cb0ddea8effa23373fff4d445a2fe700R99

**Pull request merged**